src/com/rath/rathbot/cmd/disc/ReportCmd.java

```diff
@@ -54,55 +53,46 @@ public boolean requiresDirectMessage() {
   }
   
   @Override
-  public boolean executeCommand(final IMessage msg, final String[] tokens, final int tokenDepth) {
+  public boolean executeCommand(final IMessage msg, final String[] tokens, final int tokDepth) {
     
-    IUser issuedUser = msg.getAuthor();
+    final IChannel channel = msg.getChannel();
+    final IUser issuedUser = msg.getAuthor();
     
     // Ensures at least minimum valid arguments used.
     if (tokens.length < 4) {
-      RathBot.sendDirectMessage(issuedUser, "Syntax Error! Usage: rb! report <uid> <reason>");
+      RathBot.sendDirectMessage(issuedUser, "Syntax Error! Usage: " + this.getCommandUsage());
       return RBCommand.STOP_CMD_SEARCH;
     }
     
-    long reportedUserID = 0;
-    
-    // Retrieve UID from arguments.
-    try {
-      reportedUserID = Long.parseLong(tokens[tokenDepth + 1]);
-    } catch (NumberFormatException e) {
-      e.printStackTrace();
-    }
-    
-    final IDiscordClient client = RathBot.getClient();
-    
-    // Create IUser object from REPORTED_USER_ID to reference user in messages.
-    IUser reportedUser = client.getUserByID(reportedUserID);
-    if (reportedUser == null) {
-      RathBot.sendDirectMessage(issuedUser, "Error! User not found, please enter a valid UID.");
+    // Create IUser object from token to issue disciplinary action on them.
+    final IUser infringingUser = MessageHelper.getUserFromToken(tokens[tokDepth + 1], channel);
+    if (infringingUser == null) {
+      RathBot.sendMessage(channel, "Error: Invalid UID or User not Found! "
+          + "Please verify User exists and UID is correct. " + "Remember: UIDs should only contain numbers.");
       return RBCommand.STOP_CMD_SEARCH;
     }
     
     // Message user for confirmation that report is being filed successfully.
     System.out.println("Filing report...");
-    RathBot.sendDirectMessage(issuedUser, "Thank you, your report against " + reportedUser.getName()
+    RathBot.sendDirectMessage(issuedUser, "Thank you, your report against " + infringingUser.getName()
         + " has been filed. We will look into this and take action accordingly.");
     
     Instant timestamp = msg.getTimestamp();
     
     // Log Report
-    ActionLogger.logAction(new ActionReport(timestamp, issuedUser, reportedUser));
+    ActionLogger.logAction(new ActionReport(timestamp, issuedUser, infringingUser));
     System.out.println("Report filed and logged in */logs/actions.txt");
     
-    IChannel report = client.getChannelByID(RBConfig.getReportChannelID());
+    IChannel report = RathBot.getClient().getChannelByID(RBConfig.getReportChannelID());
```

Comment:
It's my convention to make any variable that doesn't change `final`. This way, anyone looking at it can tell that it will always keep its value.